src/librustc/ty/item_path.rs

```diff
@@ -77,16 +91,21 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     /// suitable for user output. It always begins with a crate identifier.
     pub fn absolute_item_path_str(self, def_id: DefId) -> String {
         let mut buffer = LocalPathBuffer::new(RootMode::Absolute);
-        self.push_item_path(&mut buffer, def_id);
+        debug!("absolute_item_path_str: buffer={:?} def_id={:?}", buffer, def_id);
+        self.push_item_path(&mut buffer, def_id, false);
         buffer.into_string()
     }
 
     /// Returns the "path" to a particular crate. This can proceed in
     /// various ways, depending on the `root_mode` of the `buffer`.
     /// (See `RootMode` enum for more details.)
-    pub fn push_krate_path<T>(self, buffer: &mut T, cnum: CrateNum)
-        where T: ItemPathBuffer
+    pub fn push_krate_path<T>(self, buffer: &mut T, cnum: CrateNum, pushed_prelude_crate: bool)
```

Comment:
I feel like some sort of comment explaining what `pushed_prelude_crate` is used for would be nice. It feels pretty opaque right now. I'm not sure just what it should say or where it should go — but this spot seems good, as the place that *consumes* the flag.

Reply:
Added.